src/main/java/core/basesyntax/service/operation/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.service.operation;
+
+import core.basesyntax.service.FruitTransaction;
+import java.util.Map;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    @Override
+    public void handle(FruitTransaction transaction, Map<String, Integer> storage) {
+        String fruit = transaction.getFruit();
+        int amount = transaction.getAmount();
+        int currentAmount = storage.getOrDefault(fruit, 0);
+
+        if (currentAmount >= amount) {
+            storage.put(fruit, currentAmount - amount);
+        } else {
+            throw new RuntimeException("Not enough fruit in storage");
+        }
+    }
```

Comment:
Also, remember to follow the encapsulation principle. The method `handle()` is currently public, which means it can be accessed from any class. Consider reducing the access level to protected or package-private if there is no need for it to be public.